ui/test/nuxeo-collapsible.test.js

```diff
@@ -0,0 +1,158 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html, fixture, flush, isElementVisible, waitForAttrMutation, waitChanged, tap } from '@nuxeo/testing-helpers';
+import '../widgets/nuxeo-collapsible.js';
+import '../nuxeo-icons';
+/* eslint-disable no-unused-expressions */
+
+suite('nuxeo-collapsible', () => {
+  suite('Collapse behaviour', () => {
+    test('Should display nuxeo-collapsible when declaring the element', async () => {
+      const nuxeoCollapsible = await fixture(
+        html`
+          <nuxeo-collapsible></nuxeo-collapsible>
+        `,
+      );
+      await flush();
+
+      expect(isElementVisible(nuxeoCollapsible.shadowRoot.querySelector('button'))).to.be.true;
+    });
+
+    test('Should show and hide content when heading is tapped', async () => {
+      const nuxeoCollapsible = await fixture(
+        html`
+          <nuxeo-collapsible>
+            <span slot="heading">Content</span>
+            <div>Collapsed content</div>
+          </nuxeo-collapsible>
+        `,
+      );
+      await flush();
+
+      const button = nuxeoCollapsible.shadowRoot.querySelector('button');
+      const content = nuxeoCollapsible.shadowRoot.querySelectorAll('slot')[1].assignedNodes()[2];
+      const ironCollapse = nuxeoCollapsible.shadowRoot.querySelector('iron-collapse');
+      if (isElementVisible(content)) {
+        await waitChanged(ironCollapse, 'aria-hidden');
+      }
+      expect(isElementVisible(content)).to.be.false;
+
+      tap(button);
+      if (!nuxeoCollapsible.expanded) {
+        await waitForAttrMutation(nuxeoCollapsible, 'expanded', true);
+      }
+      expect(isElementVisible(content)).to.be.true;
+      expect(content.innerHTML).to.equal('Collapsed content');
+    });
+  });
+
+  suite('Properties', () => {
+    test('Should display heading when it is set', async () => {
+      const nuxeoCollapsible = await fixture(
+        html`
+          <nuxeo-collapsible>
+            <span class="heading" slot="heading">Content</span>
+          </nuxeo-collapsible>
+        `,
+      );
+      await flush();
+
+      const heading = nuxeoCollapsible.shadowRoot.querySelector('slot').assignedElements()[0];
+      expect(isElementVisible(heading)).to.be.true;
+      expect(heading.innerHTML).to.equal('Content');
+    });
+
+    test('Should display chevron when it is set', async () => {
+      const nuxeoCollapsible = await fixture(
+        html`
+          <nuxeo-collapsible chevron></nuxeo-collapsible>
+        `,
+      );
+      await flush();
+      const chevron = nuxeoCollapsible.shadowRoot.querySelector('iron-icon');
+      expect(isElementVisible(chevron)).to.be.true;
```

Comment:
Since we are the ones defining the chevron icon, I would say that it would be good to check which icon is there.
Also because we have a conditional logic for changing it and, if I'm seeing the tests correctly, we don't test it at all.

Reply:
Indeed, I will add that test